src/components/structures/auth/Registration.tsx

```diff
@@ -141,6 +141,22 @@ export default class Registration extends React.Component<IProps, IState> {
         this.replaceClient(this.props.serverConfig);
     }
 
+    componentDidUpdate(_, prevState: IState) {
+        if (prevState.doingUIAuth !== this.state.doingUIAuth) {
```

Comment:
this could still register twice, once on doingUIAuth false->true and once back (and if it keeps toggling then it would keep going

One thing to note (which I think is stupid) about addEventListener, is it does *not* check if the listener is already bound & removeEventListener only removes one instance, not all instances. So this code really needs to be solid because otherwise you'll leave stray listeners hanging around which will keep the warning even much later into the app lifecycle

Reply:
sorry to ask again, i am finding it rather confusing that addEventListener will be registered as many times the method runs so that means the `if` statement is not doing anything then cause it was supposed to only register the listener if there is a disparity between the prevState and the current State. Also if i placed the listener in componentDidMount, then that means , i 'll just put `this.state.doingUIAuth` as the if condition since there is no prevState

Reply:
> if there is a disparity between the prevState and the current State

Right but that disparity could happen multiple times within a single mounting of this component, e.g if you start registration, do captcha, then finish captcha doingUIAuth would go false->true->false then unmount

Reply:
> i 'll just put this.state.doingUIAuth as the if condition since there is no prevState

Precisely

Reply:
Yes, always register the handler, then inside `unloadCallback` you can check `doingUIAuth`

Reply:
Done